pkg/controller/autoimport/autoimport_controller.go

```diff
@@ -60,6 +71,24 @@ func (r *ReconcileAutoImport) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	mode, err := helpers.DetermineKlusterletMode(managedCluster)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
```

Comment:
I really think we should have a independent controller to handle the detached cluster :), for default managed cluster, we just return here

```
if mode == default {
  return nil
}
```

Reply:
Yep, I submitted another PR https://github.com/stolostron/managedcluster-import-controller/pull/124 implemented in the way you recommended. PTAL.